.github/workflows/ci.yaml

```diff
@@ -56,7 +56,7 @@ jobs:
     - name: Install package
       shell: bash -l {0}
       run: |
-        python -m pip install --no-deps .
```

Comment:
I removed `--no-deps` when I included the "python3.9 requires jax<=0.4.28" logic in this file. I moved the logic to `devtools/conda-envs/test_env.yaml` in 9ad2d31, but I opted to keep the removal of `--no-deps`. I don't think it really makes any difference either way, as the only dependency specified here is numpy and that specification seems to be redundant.